salt/config/__init__.py

```diff
@@ -3315,15 +3315,18 @@ def api_config(path):
     Read in the Salt Master config file and add additional configs that
     need to be stubbed out for salt-api
     '''
-    # Let's grab a copy of salt's master opts
-    opts = client_config(path, defaults=DEFAULT_MASTER_OPTS)
-    # Let's override them with salt-api's required defaults
-    api_opts = DEFAULT_API_OPTS
-    api_opts.update({
+    # Let's grab a copy of salt-api's required defaults
+    opts = DEFAULT_API_OPTS
+
+    # Let's override them with salt's master opts
+    opts.update(client_config(path, defaults=DEFAULT_MASTER_OPTS))
+
+    # Let's set the pidfile and log_file values in opts to api settings
+    opts.update({
         'pidfile': opts.get('api_pidfile', DEFAULT_API_OPTS['api_pidfile']),
         'log_file': opts.get('api_logfile', DEFAULT_API_OPTS['api_logfile']),
```

Comment:
@whiteinge Hm. Yeah I am assuming there was something going on later down the line that is using those `pidfile` and `log_file` settings somewhere in salt-api, especially since there was a change in this code recently in #37272, which removed the `log_file` setting and caused issue #38479. I didn't want to cause that bug again since that was fixed recently on this branch.

Just to make sure I am following you correctly, you're saying that the `prepend_root_dir` option also needs the `pidfile` and `log_file` settings? Something like this?
```
prepend_root_dir(opts, [
          'api_pidfile',
          'api_logfile',
     ])
```
Or am I misunderstanding what you said?